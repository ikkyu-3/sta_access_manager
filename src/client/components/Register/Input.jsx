// @flow
import type { RegisterInputProps } from "@types";

import * as React from "react";
import styles from "./style.css";

function Input(props: RegisterInputProps): React.Node {
  const { id, label, error, value, canShow, onChange } = props;
  const { inputArea, input, bad, message, hidden } = styles;

  const errorMessage = canShow ? message : `${message} ${hidden}`;
  const inputClass = error.length > 0 && canShow ? `${input} ${bad}` : input;

  return (
    <div className={inputArea}>
      <label htmlFor={id}>
        {label}
        <span className={errorMessage}>{error}</span>
        <input
          id={id}
          className={inputClass}
          type="text"
          value={value}
          onChange={onChange}
        />
      </label>
    </div>
  );
}
export default Input;
