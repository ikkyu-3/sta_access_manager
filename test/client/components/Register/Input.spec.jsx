/* eslint-disable react/jsx-boolean-value */
import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import Input from "@components/Register/Input";

const fn = () => {};

describe("Register/Input.jsxのテスト", () => {
  describe("スナップショット", () => {
    it("正しいレンダリング", () => {
      const tree = renderer.create(
        <Input
          id="id"
          label="ラベル"
          error="エラー"
          value="value"
          canShow={true}
          onChange={fn}
        />
      );
      expect(tree).toMatchSnapshot();
    });
  });

  describe("コンポーネントのテスト", () => {
    it("errorあり, canShowがtrueの時、エラーが表示される", () => {
      const input = shallow(
        <Input
          id="id"
          label="ラベル"
          error="エラー"
          value="value"
          canShow={true}
          onChange={fn}
        />
      );

      expect(input.find("span").hasClass("hidden")).toBeFalsy();
      expect(input.find("input").hasClass("bad")).toBeTruthy();
    });

    it("errorあり, canShowがfalseの時、エラーは表示されない", () => {
      const input = shallow(
        <Input
          id="id"
          label="ラベル"
          error="エラー"
          value="value"
          canShow={false}
          onChange={fn}
        />
      );

      expect(input.find("span").hasClass("hidden")).toBeTruthy();
      expect(input.find("input").hasClass("bad")).toBeFalsy();
    });

    it("input要素が変更された時、onChangeが実行される", () => {
      const mockChange = jest.fn();
      const input = shallow(
        <Input
          id="id"
          label="ラベル"
          error="エラー"
          value="value"
          canShow={false}
          onChange={mockChange}
        />
      );

      input.find("input").simulate("change", "test");
      expect(mockChange.mock.calls[0][0]).toBe("test");
    });
  });
});
