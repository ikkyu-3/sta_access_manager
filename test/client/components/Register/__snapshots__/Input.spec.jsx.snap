// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register/Input.jsxのテスト スナップショット 正しいレンダリング 1`] = `
<div
  className="inputArea"
>
  <label
    htmlFor="id"
  >
    ラベル
    <span
      className="message"
    >
      エラー
    </span>
    <input
      className="input bad"
      id="id"
      onChange={[Function]}
      type="text"
      value="value"
    />
  </label>
</div>
`;
